lib/uring/uring.ml

```diff
@@ -97,15 +102,17 @@ let realloc t iobuf =
 
 let exit {uring;_} = Uring.exit uring
 
-let with_id : type a. a t -> (Heap.ptr -> bool) -> a -> bool =
- fun t fn datum ->
-  match Heap.alloc t.user_data datum with
+let with_id_full : type a. a t -> (Heap.ptr -> bool) -> a -> extra_data:'b -> bool =
+ fun t fn datum ~extra_data ->
+  match Heap.alloc t.user_data datum ~extra_data with
```

Comment:
```suggestion
  match Heap.alloc t.data datum ~extra_data with
```

Perhaps worth renaming to indicate that this isn't just user data anymore.

Reply:
Good idea - done!